lib/xcpretty/parser.rb

```diff
@@ -193,7 +193,7 @@ module Warnings
 
       # @regex Captured groups
       # $1 = whole warning
-      GENERIC_WARNING_MATCHER = /^warning:\s(.*)$/
```

Comment:
Thanks. I understood why you guys did it that way.

There are several questions to ask oneself here:
1. How significant of a performance regression are we talking about here? I'm guessing you guys did some measurements.
2. Relying on specialized regular expressions (i.e. only look for warning: at start of line) when it comes to super important things like warnings and errors can lead to lots of pain, especially when it's customer facing and they're asking themselves wtf the build is failing. :)
3.  Take for example this change. If there was a regex that just did "warning:" that warning would not have been skipped, and I would really not have needed this pull request.